### PR TITLE
Referral program v2 next fee operations

### PR DESCRIPTION
**Pull request by ochaloup**
Adding processing of fees for two other operations and changing (adding) the tests appropriatelly.

### CODE REVIEW

**Comment by ochaloup** [comment]
@luciotato one thing which is "problematic" and I don't know how to tackle it.
The tests are dependent on defined path on the local machine that points to `marinade-anchor` repository.
https://github.com/marinade-finance/liquid-staking-referral-program/blob/referral-program-v2/programs/marinade-referral/tests/Cargo.toml#L11
(that's how it was and it could be the reason of why the tests were in a separate repo prior of my changes?

**Comment by ochaloup** [comment]
I added tests to check that passing wrong referral deposit msol account fails.

File: programs/marinade-referral/src/instructions/deposit_sol.rs [hunk 1]
```diff
@@ -29,7 +28,7 @@ pub struct Deposit<'info> {
     pub system_program: AccountInfo<'info>,
```
Comment (luciotato) [inline programs/marinade-referral/src/instructions/deposit_sol.rs:28]: Note to self: add comment that system_program and token_program are checked by Marinade core in the cpi call

File: programs/marinade-referral/src/instructions/liquid_unstake.rs [hunk 2]
```diff
@@ -57,7 +72,7 @@ impl<'info> LiquidUnstake<'info> {
         };
 
         // compute fee in msol
-        let msol_fee = liquid_unstake_fee.apply(msol_amount);
+        let msol_fee = liquid_unstake_fee.apply(msol_amount_fee_deducted);
         msg!("msol_fee {}", msol_fee);
```
Comment (luciotato) [inline programs/marinade-referral/src/instructions/liquid_unstake.rs:76]: comment that's liquid-unstake msol fee (not partner fee)

**Review by luciotato — APPROVED** [review]

**Pull request merged**